src/test/groovy/com/github/rodionovsasha/commentservice/controllers/CommentControllerTest.groovy

```diff
@@ -66,8 +73,73 @@ class CommentControllerTest extends Specification {
         response.contentLength == 0
     }
 
+    def "#add creates a new comment"() {
+        when:
+        def response = extractJson(addComment([content: DEFAULT_CONTENT], TOPIC_ID, USER_ID), HttpStatus.CREATED)
+
+        then:
+        1 * service.add(DEFAULT_CONTENT, TOPIC_ID, USER_ID) >> defaultComment
+        with(response) {
+            id == 0
+            content == "Why you little...!"
+            !archived
+            date instanceof Long
+        }
+    }
+
+    def "#add does not create a new comment when user does not exist"() {
+        given:
+        service.add(DEFAULT_CONTENT, TOPIC_ID, USER_ID) >> { throw UserNotFoundException.forId(USER_ID) }
+
+        when:
+        def response = addComment([content: DEFAULT_CONTENT], TOPIC_ID, USER_ID)
+
+        then:
+        extractJson(response, HttpStatus.NOT_FOUND) == [code: 404, message: "The user with id '1' could not be found"]
+    }
+
+    def "#add does not create a new comment when user is not active"() {
+        given:
+        service.add(DEFAULT_CONTENT, TOPIC_ID, USER_ID) >> { throw InactiveUserException.forId(USER_ID) }
+
+        when:
+        def response = addComment([content: DEFAULT_CONTENT], TOPIC_ID, USER_ID)
+
+        then:
+        extractJson(response, HttpStatus.FORBIDDEN) == [code: 403, message: "The user with id '1' is not active"]
+    }
+
+    def "#add does not create a new comment when topic is archived"() {
+        given:
+        service.add(DEFAULT_CONTENT, TOPIC_ID, USER_ID) >> { throw ArchivedTopicException.forId(TOPIC_ID) }
+
+        when:
+        def response = addComment([content: DEFAULT_CONTENT], TOPIC_ID, USER_ID)
+
+        then:
+        extractJson(response, HttpStatus.INTERNAL_SERVER_ERROR) == [code: 500, message: "The topic with id '1' is archived"]
+    }
+
+    def "#add does not create a new comment when content is empty"() {
+        when:
+        def response = extractJson(addComment([content: ""], TOPIC_ID, USER_ID), HttpStatus.INTERNAL_SERVER_ERROR)
+
+        then:
+        0 * service.add("", TOPIC_ID, USER_ID)
+        with(response) {
+            code == 500
```

Comment:
Not critical for a demo project, but looking at the error it should be probably BAD_REQUEST